language/move-stdlib/src/natives/debug.rs

```diff
@@ -28,62 +545,41 @@ pub struct PrintGasParameters {
     pub base_cost: InternalGas,
 }
 
-#[cfg(feature = "testing")]
-fn is_string_type(context: &NativeContext, ty: &Type, move_std_addr: AccountAddress) -> bool {
-    let ty_tag = match context.type_to_type_tag(ty) {
-        Ok(ty_tag) => ty_tag,
-        Err(_) => return false,
-    };
-
-    match ty_tag {
-        TypeTag::Struct(struct_tag) => {
-            struct_tag.address == move_std_addr
-                && struct_tag.module.as_str().eq("string")
-                && struct_tag.name.as_str().eq("String")
-        }
-        _ => false,
-    }
-}
-
-#[allow(unused_mut)]
 #[inline]
 fn native_print(
     gas_params: &PrintGasParameters,
     _context: &mut NativeContext,
     mut ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
-    move_std_addr: AccountAddress,
+    _move_std_addr: AccountAddress,
```

Comment:
```suggestion
    move_std_addr: AccountAddress,
```

Nit: the Rust compiler will avoid emitting unused variable warnings if the name has a `_` prefix, so it is safest to only give a variable name this prefix if you are not using it.

Reply:
Yes, indeed! But the problem with leaving it as `move_std_addr` is it emits the warning when the `testing` feature is off... because nobody uses it in that case.